src/shape.rs

```diff
@@ -213,8 +216,8 @@ impl Shape {
         }
     }
 
-    pub(crate) fn block_left(&self, width: usize) -> Option<Shape> {
-        self.block_indent(width).sub_width(width)
+    pub(crate) fn block_left(&self, n: usize, span: Span) -> Result<Shape, ExceedsMaxWidthError> {
+        self.block_indent(n).sub_width(n, span)
```

Comment:
I had done that initially and it tripped me up because `.map(|width| Shape { width, ...` shadows the argument. I wanted to change it because the field `width` and the argument `width` are totally different things. Of course I can work around that though.

Reply:
I'd be fine naming it something other than `width`, but I think it would be nice if it were more descriptive than `n`. I'm open to suggestions.

Reply:
Offset is good. Or amount, distance, delta.

Reply:
Any of those work for me. I'll leave it up to you on which one you think would work best.

Reply:
I went with delta. Seems precise and short!